### PR TITLE
{Packaging} Remove Jinja2, MarkupSafe and pycparser from requirements.txt

### PR DESCRIPTION
**Pull request by bebound**
`Jinja2` and `MarkupSafe` are not used in CLI.
`pycparser` is a dependency of `cffi`.
We don't need to include them in requirements.txt

Resolve https://github.com/Azure/azure-cli/issues/29421

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

**Comment by jiasli** [comment]
I searched the whole repo and neither `MarkupSafe` nor `pycparser` is directly used. Maybe we can remove them?

**Comment by bebound** [comment]
Yes. In this PR I wrote that jinja2 and markupsafe can be removed and I totally forget it.  https://github.com/Azure/azure-cli/pull/27717

File: src/azure-cli/azure/cli/command_modules/feedback/tests/latest/test_feedback.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 logger = logging.getLogger(__name__)
 
 
+@unittest.skip('alias extension is not working: https://github.com/Azure/azure-cli/issues/29422')
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/feedback/tests/latest/test_feedback.py:32]: After removing `markupsafe`, the `alias` extension does not work anymore. Previously, the CLI's `markupsafe` was loaded; now it loads the outdated alias's `markupsafe` package.
```
az alias create -n foo123 -c bar123
Unable to load extension 'alias: cannot import name 'soft_unicode' from 'markupsafe' (C:\Users\kk\.azure\cliextensions\alias\markupsafe\__init__.py)'. Use --debug for more information.
```

**Review by jiasli — APPROVED** [review]

**Pull request merged**